src/bump-timestamp

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/python3 -u
+
+'''
+    This script is called by `build` command and updates builds timestamp.
```

Comment:
It has to rescan it as FCOS pipeline (test builds at least) don't fetch previous installs.

Maybe it should only rescan if no builds founds?

Reply:
> It has to rescan it as FCOS pipeline (test builds at least) don't fetch previous installs.

Heh, I think that's an argument for *not* rescanning. :)

If we rescan, then the `builds.json` becomes useless because we only pull down the last build, so we'd drop all the other builds in the file (of course, all this is moot today since we don't actually re-upload `builds.json` after no-op builds... but we *could*).

Reply:
If you want to test this logic in CI, you can use `cosa build ostree --force-image` as a way to get more builds out to play with without increasing test times by much. E.g.:

```diff
diff --git a/.cci.jenkinsfile b/.cci.jenkinsfile
index 0a321ab..62deed1 100644
--- a/.cci.jenkinsfile
+++ b/.cci.jenkinsfile
@@ -33,6 +33,16 @@ coreos.pod([image: 'registry.fedoraproject.org/fedora:30', privileged: true, kvm
               cosa_cmd("buildextend-vmware")
               cosa_cmd("compress")
               cosa_cmd("buildupload --dry-run s3 --acl=public-read my-nonexistent-bucket/my/prefix")
+
+              // test pruning
+              cosa_cmd("build ostree --force-image")
+              cosa_cmd("build ostree --force-image")
+              cosa_cmd("build ostree --force-image")
+              // check that only three builds are kept
+              // ...
+              cosa_cmd("bump-timestamp")
+              // check that `timestamp` is updated
+              // ...
           }
       }
 }
```

Though feel free to adapt as you'd like!

Reply:
I still think `bump-timestamp` shouldn't rescan builds as mentioned in https://github.com/coreos/coreos-assembler/pull/839/files#r335999198. `--skip-prune` should really just be about inserting the new build and leaving everything else untouched. This is important for pipelines.

> What if we add --insert-only part to bump-timestamp and don't call prune during build?

I think we should still prune though if `--skip-prune` isn't specified.

OK, how about just having a tiny `insert_build` script that `build --skip-prune` uses to just inject the new build at the top of the list? Then for `bump-timestamp`, we either drop it (since nothing really needs it right now), or at the very least make it purely about, well, bumping the timestamp. :)

Reply:
> The issue is that CI run is not fetching previous builds so I can't verify it is in fact working

I noticed you were playing with adding tests here, but don't see any now. Why not keep them?

Reply:
>I noticed you were playing with adding tests here, but don't see any now. Why not keep them?

oops, forcepushed the wrong branch, brought those back

Reply:
@jlebon PTAL

>I think we should still prune though if --skip-prune isn't specified.

Done

>OK, how about just having a tiny insert_build script that build --skip-prune uses to just inject the new build at the top of the list?

This is now what `bump-timestamp` does. Perhaps it should be renamed to `insert-build`?

Note that `prune --dry-run` still mutates `builds.json` - I'm not sure if this is the right thing to do

Reply:
> This is now what bump-timestamp does. Perhaps it should be renamed to insert-build?

Yup, SGTM. :+1: 
Though let's make it an internal helper thing instead of a top-level command. Honestly, you could just get away with just calling python inline like here: https://github.com/coreos/coreos-assembler/blob/53f8a50595377098797d9c60cfa0c51dcd86d8ab/src/cmdlib.sh#L659

> Note that prune --dry-run still mutates builds.json - I'm not sure if this is the right thing to do

Hmm, yeah that doesn't sound right. We can just gate the `builds.bump_timestamp()` call on `args.dry_run`, right?

Reply:
>Though let's make it an internal helper thing instead of a top-level command.

Okay, implemented, although that excludes this function from pylint checks, so I'm entirely sure this is a good idea in the long run